safekeeper/src/handler.rs

```diff
@@ -140,6 +140,12 @@ impl<IO: AsyncRead + AsyncWrite + Unpin + Send> postgres_backend::Handler<IO>
                 }
             }
 
+            let ttid = TenantTimelineId::new(
+                self.tenant_id.unwrap_or(TenantId::from([0u8; 16])),
+                self.timeline_id.unwrap_or(TimelineId::from([0u8; 16])),
+            );
+            tracing::Span::current().record("ttid", tracing::field::display(ttid));
```

Comment:
is this not quite confusing if the span comes from caller somewhere. it would be better to fill in the blanks near the span creation...

Reply:
Oki, found it near the end of the changes, but this as a understanding code.

As a related idea for keeping the spans up to date, check out pageserver's `debug_assert_current_span_has_tenant_id` -- you could easily create one for `ttid` and use that to enforce that the span parameter is present (but not yet detect if it is Empty).

Reply:
That's not always possible. In this case, we need to establish postgres protocol connection and read first StartupMessage to get ttid from it.